wordpress-mobile/WordPress-iOS
# Today widgets: fix translations

Fixes #n/a

When the `All Time` and `This Week` widgets were implemented, the strings were never actually translated. Per @jkmassel "The problem (I think) is that while we have a way to combine .strings files (ie – to translate strings in a dependency), we don't have the ability to handle more than one target". Since the Today widget already existed, this is the one target that was handled and the strings translated.

After discussing solutions with @jkmassel , for now we'll share the WP `Localizable.strings` with the widgets so they are translated. Even though Today was being translated, for consistency, I've removed the widget translations and it's also now using the WP `Localizable.strings`. 

Note this only affects the strings displayed in the widgets, not the actual widget names.

To recap, the changes:
- `WordPressTodayWidget` - removed localization files.
- `WordPress\Resources\Localizable.strings` - added to 3 widget targets.

<img width="251" alt="Screen Shot 2020-10-28 at 2 54 25 PM" src="https://user-images.githubusercontent.com/1816888/97495704-90e3c100-192d-11eb-9e4c-be2ae29c049b.png">

---
_Tip_: sometimes the widgets get... confused... and will display an 'unable to load' message. You can fix this by either:
- Build the widget scheme first, then run the `WordPress` scheme.
- Run the widget scheme directly.

<img width="294" alt="Screen Shot 2020-10-28 at 3 21 02 PM" src="https://user-images.githubusercontent.com/1816888/97498136-43695300-1931-11eb-93ce-81dc909f8622.png">

**To test:**

- Run the app.
- Add the widgets to the Today View.
- Change the device language.
- Verify the text is translated.

| ![unconfigured-english](https://user-images.githubusercontent.com/1816888/97496640-dc4a9f00-192e-11eb-948a-5634c1530296.jpg) | ![unconfigured-spanish](https://user-images.githubusercontent.com/1816888/97496649-e076bc80-192e-11eb-8e8e-91c4ca27d50e.jpg) |
|--------|-------|

- In the app, set a site to use for the Widgets. (Stats > Widgets > Use this site)

![widget_site](https://user-images.githubusercontent.com/1816888/97496924-524f0600-192f-11eb-939c-a16d7708c289.jpg)

- Return to the Today View. 
- Verify the widget text is translated.

| ![configured-english](https://user-images.githubusercontent.com/1816888/97497548-49126900-1930-11eb-9769-ee37ca4b4d0f.jpg) | ![configured-spanish](https://user-images.githubusercontent.com/1816888/97497565-52033a80-1930-11eb-828b-671a2810a1ba.jpg) |
|--------|-------|

---
PR submission checklist:
